haystack/preview/dataclasses/document.py

```diff
@@ -110,27 +82,49 @@ class Document:
-    so if the file is moved, the hash is different, but if the file is modified without renaming it, the has will
-    not differ.
+    :param id: Unique identifier for the document. Do not provide this value when initializing a document: it will be
+        generated based on the document's attributes (see id_hash_keys).
+    :param text: Text of the document, if the document contains text.
+    :param array: Array of numbers associated with the document, if the document contains matrix data like image,
+        audio, video, and such.
+    :param dataframe: Pandas dataframe with the document's content, if the document contains tabular data.
+    :param blob: Binary data associated with the document, if the document has any binary data associated with it.
+    :param mime_type: MIME type of the document. Defaults to "text/plain".
+    :param metadata: Additional custom metadata for the document.
+    :param id_hash_keys: List of keys to use for the ID hash. Defaults to the four content fields of the document:
+        text, array, dataframe and blob. This field can include other document fields (like mime_type) and metadata's
+        top-level keys. Note that the order of the keys is important: the ID hash will be generated by concatenating
+        the values of the keys in the order they appear in this list. Changing the order impacts the ID hash.
+    :param score: Score of the document. Used for ranking, usually assigned by retrievers.
+    :param embedding: Vector representation of the document.
     """
 
     id: str = field(default_factory=str)
-    content: Any = field(default_factory=lambda: None)
-    content_type: ContentType = "text"
+    text: Optional[str] = field(default=None)
+    array: Optional[numpy.ndarray] = field(default=None)
+    dataframe: Optional[pandas.DataFrame] = field(default=None)
+    blob: Optional[bytes] = field(default=None)
+    mime_type: str = field(default="text/plain")
     metadata: Dict[str, Any] = field(default_factory=dict, hash=False)
-    id_hash_keys: List[str] = field(default_factory=list, hash=False)
+    id_hash_keys: List[str] = field(default_factory=lambda: ["text", "array", "dataframe", "blob"], hash=False)
     score: Optional[float] = field(default=None, compare=True)
     embedding: Optional[numpy.ndarray] = field(default=None, repr=False)
 
     def __str__(self):
-        return f"{self.__class__.__name__}('{self.content}')"
+        if self.text:
+            return f"{self.__class__.__name__}(mimetype: {self.mime_type}, text: '{self.text}')"
+        if self.array:
+            return f"{self.__class__.__name__}(mimetype: {self.mime_type}, array: '{self.array}')"
+        if self.dataframe:
+            return f"{self.__class__.__name__}(mimetype: {self.mime_type}, dataframe: '{self.dataframe}')"
+        if self.blob:
+            return f"{self.__class__.__name__}(mimetype: {self.mime_type}, binary only)"
```

Comment:
They can do that, I don't think it's an issue. Like a video: it could have a blob (the binary file) an array (the decoded binary) and text (the transcription).